src/meta/http/MetaHttpReplaceHostHandler.cpp

```diff
@@ -156,12 +171,54 @@ bool MetaHttpReplaceHostHandler::replaceHost(std::string ipv4From, std::string i
   kvstore_->asyncMultiPut(
       kDefaultSpaceId, kDefaultPartId, std::move(data), [&](nebula::cpp2::ErrorCode code) {
         updateSucceed = (code == nebula::cpp2::ErrorCode::SUCCEEDED);
-        errMsg_ = folly::stringPrintf("write to kvstore failed, %s , %d", __func__, __LINE__);
+        errMsg_ = folly::stringPrintf("Write to kvstore failed, %s , %d", __func__, __LINE__);
         baton.post();
       });
   baton.wait();
   return updateSucceed;
 }
 
+bool MetaHttpReplaceHostHandler::replaceHostInZone(std::string ipv4From, std::string ipv4To) {
+  folly::SharedMutex::WriteHolder wHolder(LockUtils::spaceLock());
+  const auto& zonePrefix = MetaKeyUtils::zonePrefix();
+  std::unique_ptr<kvstore::KVIterator> iter;
+  auto kvRet = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, zonePrefix, &iter);
+  if (kvRet != nebula::cpp2::ErrorCode::SUCCEEDED) {
+    errMsg_ = folly::stringPrintf("Can't get zone prefix=%s", zonePrefix.c_str());
+    LOG(INFO) << errMsg_;
+    return false;
+  }
+
+  std::vector<nebula::kvstore::KV> data;
+  while (iter->valid()) {
+    bool needUpdate = false;
+    auto zoneName = MetaKeyUtils::parseZoneName(iter->key());
+    auto hosts = MetaKeyUtils::parseZoneHosts(iter->val());
+    for (auto& host : hosts) {
+      if (host.host == ipv4From) {
+        host.host = ipv4To;
+        needUpdate = true;
```

Comment:
break out?

Reply:
Better check that, I do not know the constraints on hosts.

Reply:
After checking, I'll we could not add break. Because multi-services could lay in the same host, in which they share the same ip.